wasmVirtualMachine.js

```diff
@@ -423,18 +471,85 @@ function build_module(byte_code) {
                 i++;
                 decode = Leb.decodeUint32(byte_code, i);
                 i = decode.nextIndex;
-                size = decode.value;
+                expected_end = i + decode.value;
+
+                // get array of globals
+                decode = Leb.decodeUint32(byte_code, i);
+                i = decode.nextIndex;
+                num_els = decode.value;
+
+                // get global
+                for (let j = 0; j < num_els; j++) {
+                    // get global description (type and mutability, mut is 0 or 1)
+                    decode = Leb.decodeUint32(byte_code, i);
```

Comment:
The value type is a byte not a uint_32
https://webassembly.github.io/spec/binary/types.html#binary-globaltype